src/Middleware/HostFiltering/src/HostFilteringMiddleware.cs

```diff
@@ -16,14 +15,14 @@ namespace Microsoft.AspNetCore.HostFiltering;
 public class HostFilteringMiddleware
 {
     // Matches Http.Sys.
-    private static readonly byte[] DefaultResponse = Encoding.ASCII.GetBytes(
-        "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.01//EN\"\"http://www.w3.org/TR/html4/strict.dtd\">\r\n"
-            + "<HTML><HEAD><TITLE>Bad Request</TITLE>\r\n"
-            + "<META HTTP-EQUIV=\"Content-Type\" Content=\"text/html; charset=us-ascii\"></ HEAD >\r\n"
-            + "<BODY><h2>Bad Request - Invalid Hostname</h2>\r\n"
-            + "<hr><p>HTTP Error 400. The request hostname is invalid.</p>\r\n"
-            + "</BODY></HTML>"
-    );
+    private static readonly byte[] DefaultResponse = (
```

Comment:
Looks like this can change to `private static ReadOnlySpan<byte> => ...` and get rid of the `ToArray` call.
The usage code can then use the `ReadOnlySpan` overload of `WriteAsync`.